src/components/shared/page.tsx

```diff
@@ -129,6 +135,25 @@ const getImageMappings = (imageSalt: string, capi: Content): ImageMappings => {
         return { ...blockMappings, ...getContributorMappings(capi.tags, imageSalt) }
 }
 
+const liveblogHydrationProps = (capi: Content): Content => {
+    const { id: contentId, type, webTitle, webUrl, apiUrl, fields, tags, references, isHosted, blocks } = capi;
+    const { main, body, totalBodyBlocks } = blocks as Blocks;
+    const { createdBy, lastModifiedBy, ...mainBlocks } = main as Block;
```

Comment:
What happens if `blocks` and/or `main` are `undefined`?

Reply:
refactored to avoid casting